pkg/build/controller/config_controller.go

```diff
@@ -6,19 +6,38 @@ import (
 	"github.com/golang/glog"
 	kapi "k8s.io/kubernetes/pkg/api"
 	kerrors "k8s.io/kubernetes/pkg/api/errors"
-	"k8s.io/kubernetes/pkg/util"
+	kutil "k8s.io/kubernetes/pkg/util"
 
 	buildapi "github.com/openshift/origin/pkg/build/api"
 	buildclient "github.com/openshift/origin/pkg/build/client"
+	"github.com/openshift/origin/pkg/build/util"
 )
 
 type BuildConfigController struct {
 	BuildConfigInstantiator buildclient.BuildConfigInstantiator
+	BuildConfigDeleter      buildclient.BuildConfigDeleter
+	BuildDeleter            buildclient.BuildDeleter
+	BuildLister             buildclient.BuildLister
 }
 
 func (c *BuildConfigController) HandleBuildConfig(bc *buildapi.BuildConfig) error {
 	glog.V(4).Infof("Handling BuildConfig %s/%s", bc.Namespace, bc.Name)
 
+	if !bc.DeletionTimestamp.IsZero() {
```

Comment:
We could set the grace period as very large. Then, in the BC controller, once we know all builds are deleted, we could shorten the period. I'm not 100% sure this could be done from the controller, but I'll go check.